tests/e2e/tidbcluster/serial.go

```diff
@@ -752,6 +753,16 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
 				if err != nil {
 					return false, nil
 				}
+				t, ok := tac.Annotations[label.AnnTiKVReadyToScaleTimestamp]
+				if !ok {
+					klog.Info("tac has no tikv.tidb.pingcap.com/ready-to-scale-timestamp annotation")
+					return false, nil
+				}
+				readyToScaleTimestamp, err := strconv.ParseInt(t, 10, 64)
+				if err != nil {
+					framework.ExpectNoError(err, "tikv.tidb.pingcap.com/ready-to-scale-timestamp parse int error")
+					return false, nil
+				}
```

Comment:
I think we shouldn't check `AnnTiKVReadyToScaleTimestamp` and whether tikv replicas is 4 in the same `Wait.Pull`, we should first check `AnnTiKVReadyToScaleTimestamp` annotation and tidbclusterAutoScaler status in one `Wait.Pull`, then we would check whether `tidbclusterAutoScaler` become ready Normal and whether tikv replicas become 4.